### ISSUE TITLE
Trouble running scaled down OC4 test case

### ISSUE THREAD

**Issue by SudharsanKalidoss** [OpenFAST/openfast#800]
Hi Everyone,

   I trying to run OC4 test case with scaled down parameters. I used the scaling ratio of 1:50. I reduced the simulation time step to 0.0018 from 0.0125 which is used for full scale model. I get the following error.
![Scale_down_dt_0018](https://user-images.githubusercontent.com/17265645/127696864-f321b595-ae6c-4e5e-963c-5d323dbcf19c.png)

I further reduced the time step to 0.0015. The simulation starts and it hangs. It doesn't show any error or proceeds. I'm not sure where I'm going wrong. Can anyone help me on this?
![Scale_down_dt_0015](https://user-images.githubusercontent.com/17265645/127697161-ca8b79f3-de90-475e-9a1f-3a19afdcfd44.png)

**Comment by jjonkman** [comment]
Dear @SudharsanKalidoss,

Are you applying scaling laws for mass, stiffness, geometry, time, etc.?  Do you derive the time step from the scaling law?  Are you sure you've applied the scaling laws correctly to all parameters?  Are you also changing constants within the controller?  

In your first example, it sounds like the model is not balanced in the vertical direction (weight + vertical mooring pretension does not equal the buoyancy) or the model is numerically unstable.  I'm not sure I understand why OpenFAST would ever just hang (I've not heard of such an issue before), but there are many unexpected warnings before that anyway.  Both indicate a problem with the simulation set up.

Best regards.

**Comment by SudharsanKalidoss** [comment]
Dear @jjonkman,

   I used the scaling law as suggested in the following paper. I applied scaling law to all the parameters including mass, inertia, stiffness, geometry, and time. 
[FAST code verification of scaling laws for DeepCwind Floating Wind System Tests.pdf](https://github.com/OpenFAST/openfast/files/6909695/FAST.code.verification.of.scaling.laws.for.DeepCwind.Floating.Wind.System.Tests.pdf)

Is it possible to avail the modeling parameters for DeepCwind floating wind system which is used for scaled down experiment?

Is there something else do I need to do to balance weight and vertical mooring pretension other than applying scaling law?

Thanks

**Comment by jjonkman** [comment]
Dear @SudharsanKalidoss,

The scaling laws presented in this paper, if properly applied to all variables, should keep the system stable.  In fact, as that paper shows, the results are identical if you run the simulation at model scale and scale the results back up to full scale versus running the simulation at full scale.  For this reason, we rarely build an OpenFAST model at model scale.

From the paper, time is scaled by lambda^(0.5), so for lambda = 1/50 and a full-scale DT of 0.0125 s, model-scale DT of 0.00177 s should be sufficient.  I would guess something in your model has not been consistently scaled.  Unfortunately, I'm not sure where the FAST model used in this paper is stored, so, I can't share it with you.  I would suggest simplifying the model to debug, e.g., does the model work as expected if you model the 6 degree of freedom rigid-body motion of the floater excited only be waves?

Best regards,

**Comment by SudharsanKalidoss** [comment]
Dear @jjonkman,

   I tried running model with only HydroDyn enable. It was working fine. Then enabled moorDyn I encountering the following error. It seems like there is a format error for the moorDyn input file. 
![Hydro_moor](https://user-images.githubusercontent.com/17265645/127888104-29f7050c-a2eb-4659-8885-55573d57d6eb.png)

Thanks

**Comment by jjonkman** [comment]
Dear @SudharsanKalidoss,

You say that you enabled MoorDyn, but the error message is related to MAP++.  Have you set CompMooring to 1 (MAP++) or 3 (MoorDyn) and is your mooring input file formatted for MAP++ or MoorDyn?

Best regards,

**Comment by SudharsanKalidoss** [comment]
Dear @jjonkman,

    I'm sorry. I didn't notice that. I changed it and MoorDyn solver ran. It shows a different error.
![Hydro_moor_2](https://user-images.githubusercontent.com/17265645/127909530-d2022e6e-e5af-4215-b623-f02db1ed00ae.png)

Thanks

**Comment by jjonkman** [comment]
Dear @SudharsanKalidoss,

Well, now it looks like your simulation has gone numerically unstable very quickly...within the first few time steps.  I would guess you still some scaled inputs set incorrectly.  Are you running the simulation now with only ElastoDyn, HydroDyn, and MoorDyn enabled and only the platform DOFs enbled in ElastoDyn?  Is the solution what you expect when you only have one platform DOF enabled?  Does a particular platform DOF bring about the instability?

Best regards,

**Comment by SudharsanKalidoss** [comment]
Dear @jjonkman,

   Yes, I'm running the simulation with only ElastoDyn, HydroDyn and MoorDyn enabled. I enabled only the platform DOFs. I tried running the setup by enabling the DOFs one by one. The simulation runs fine with 5 DOFs (except yaw DOF). Including yaw DOF diverges the simulation. Combination of yaw and sway DOFs works fine. COmbination of yaw with any other DOFs diverges.
![Hydro_moor_yaw](https://user-images.githubusercontent.com/17265645/127922467-ea0cd9c5-687a-47de-af2b-fde35f3b8d2a.png)

Thanks
